nova/virt/vmwareapi/vm_util.py

```diff
@@ -1515,79 +1515,176 @@ def _get_host_reservations_map(groups=None):
+            CONF.vmware.memory_reservation_max_ratio_fallback
+
+    return {
+        "vcpus": total_vcpus,
+        "vcpus_used": total_vcpus_used,
+        "vcpus_reserved": total_vcpus_reserved,
+        "max_vcpus_per_host": max_vcpus_per_host,
+        "memory_mb": total_memory_mb,
+        "memory_mb_used": total_memory_mb_used,
+        "memory_mb_reserved": total_memory_mb_reserved,
+        "max_mem_mb_per_host": max_mem_mb_per_host,
+        "vm_reservable_memory_ratio": vm_reservable_memory_ratio,
+        "cpu_info": (previous_cpu_info if cpu_infos_equal
+                     else "CPUs for this cluster have different values!")
+    }
+
+
+def _host_props_to_cpu_info(host_props):
+    processor_type = None
+    cpu_vendor = None
+    hardware_cpu_pkg = host_props.get("hardware.cpuPkg")
+    if hardware_cpu_pkg and hardware_cpu_pkg.HostCpuPackage:
+        t = hardware_cpu_pkg.HostCpuPackage[0]
+        processor_type = t.description
+        cpu_vendor = t.vendor.title()
+
+    features = []
+    if "config.featureCapability" in host_props:
+        feature_capability = host_props["config.featureCapability"]
+        for feature in feature_capability.HostFeatureCapability:
+            if not feature.featureName.startswith("cpuid."):
+                continue
+            if feature.value != "1":
+                continue
+
+            name = feature.featureName
+            features.append(name.split(".", 1)[1].lower())
+    cpu_info = {
+        "model": processor_type,
+        "vendor": cpu_vendor,
+        "features": sorted(features)
+    }
+    hardware_cpu_info = host_props.get("hardware.cpuInfo")
+    if hardware_cpu_info:
+        cpu_info["topology"] = {
+            "cores": hardware_cpu_info.numCpuCores,
+            "sockets": hardware_cpu_info.numCpuPackages,
+            "threads": hardware_cpu_info.numCpuThreads
+        }
+    return cpu_info
+
+
+def _set_hypervisor_type_and_version(stats, host_props):
+    product = host_props.get("summary.config.product")
+    if not product:
+        return
+
+    stats["hypervisor_type"] = product.name
+    stats["hypervisor_version"] = convert_version_to_int(product.version)
+
+
+def _process_host_stats(obj, host_reservations_map):
+    host_props = propset_dict(obj.propSet)
+    runtime_summary = host_props["summary.runtime"]
+    hardware_summary = host_props["summary.hardware"]
+    stats_summary = host_props["summary.quickStats"]
+    # Total vcpus is the sum of all pCPUs of individual hosts
+    # The overcommitment ratio is factored in by the scheduler
+    threads = hardware_summary.numCpuThreads
+    mem_mb = hardware_summary.memorySize // units.Mi
+
+    stats = {
+        "available": (not runtime_summary.inMaintenanceMode and
+                      runtime_summary.connectionState == "connected"),
+        "vcpus": threads,
+        "vcpus_used": 0,
+        "memory_mb": mem_mb,
+        "memory_mb_used": stats_summary.overallMemoryUsage,
```

Comment:
Is this what we talked about in a previous PR that "updates the compute node in the DB with live values"? "vpus_used" is 0, so we could not rely on that, and do you know if the `overallMemoryUsage` counts active memory only, i.e. could report a way too low value compared to configured memory?

Reply:
> Is this what we talked about in a previous PR that "updates the compute node in the DB with live values"?

Yes, that would be the code.

> "vpus_used" is 0, so we could not rely on that

Yes, you are right. The `vcpus_used` value is fairly useless. I can change that, but this is the way how it is also reported in the stable branch.

>   do you know if the overallMemoryUsage counts active memory only, i.e. could report a way too low value compared to configured memory?

That is the actual memory used on the host, so yes, it will be lower than configured value. I think, there is no host stat for configured memory.

Reply:
I would though prefer to fix `vcpus_used` in a separate pull request.

Reply:
> Is this what we talked about in a previous PR that "updates the compute node in the DB with live values"?

Yes, it is (or more exactly, that is the code which mimics the same behaviour we currently have).

> "vpus_used" is 0, so we could not rely on that

Indeed. I can fix that. But I would prefer to do that in a separate pull-request. Question though is, if the work is needed. If we move forward with an external scheduling service which pulls its own metrics, then the value doesn't matter much.

> do you know if the `overallMemoryUsage` counts active memory only, i.e. could report a way too low value compared to configured memory?

Yes, it is only actually used memory on the host, so it lower (at times significant lower) value than the configured memory.
I would though not say that it is too low, as from my point-of-view it serves its own purpose. If you want to know the configured memory, that you can get from the placement-api, and it probably makes sense to take both values into consideration when scheduling a VM.

Reply:
My point here was more into the direction of freeing up a BB. As I understood it, we would take the "actually used" RAM, which might be significantly less that placement would allow. I was wondering at which point this would fail - or would it even fail?

Reply:
Regardless if we go through the scheduler or not (by forcing the host), the migration would still call to the placement-api to 
claim the resources on the destination host/node for the migration. If we do not have enough resources, I would expect the placement-api to reject the claim. For the live migration, it would fail more or less here: https://github.com/sapcc/nova/blob/stable/rocky-m3/nova/conductor/tasks/live_migrate.py#L89-L116.
For cold-migration (resize) the code is here: https://github.com/sapcc/nova/blob/stable/rocky-m3/nova/conductor/tasks/migrate.py#L235-L280

Reply:
The planning in `screw` is not intended to be a proper solution and needs a replacement eventually. But it is also a different topic.

Reply:
It is indeed. That's why I suggested to continue this discussion in slack. It just feels like it would fail more often than not if it takes those values as a base.